tests/testthat/test-as-data-frame.R

```diff
@@ -77,15 +77,29 @@ with_mock_crunch({
             as.vector(ds$gender))
     })
 
+    test_that("coerceVariable produces expected list entries", {
+        df <- lapply(ds, as.vector)
+        names(df) <- names(ds)
+        df[] <- lapply(df, as.character)
+        df <- structure(df, class = "data.frame", row.names=c(NA, -nrow(ds)))
+        expect_is(coerceVariable(ds$birthyr, df)[[1]], "numeric")
+        expect_is(coerceVariable(ds$location, df)[[1]], "factor")
+        expect_is(coerceVariable(ds$textVar, df)[[1]], "character")
+        expect_is(coerceVariable(ds$starttime, df)[[1]], "Date")
+        expect_is(coerceVariable(ds$mymrset, df)[[1]], "factor")
+        expect_is(coerceVariable(ds$catarray, df)[[1]], "factor")
```

Comment:
Should this be a factor? I remember a discussion about how these should end up and if I'm remembering correctly: arrays should have multiple columns, with the subvariable name/alias being the column name.

Reply:
I think it should, when you pass a MR variable with three subvars to `coerceVariable` it will return a list of length 3 with each entry being a single factor variable with the alias of the subvar as the name of the variable.